includes/REST_API/Stories_Media_Controller.php

```diff
@@ -69,6 +69,33 @@ public function get_items( $request ) {
 		return $response;
 	}
 
+	/**
+	 * Updates a single attachment.
+	 *
+	 * Override the existing method so we can set parent id.
+	 *
+	 * @since 1.2.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, WP_Error object on failure.
+	 */
+	public function update_item( $request ) {
+
+		if ( $request['web_stories_parent'] && get_post( $request['web_stories_parent'] ) ) {
+			$args   = [
+				'ID'          => $request['id'],
+				'post_parent' => $request['web_stories_parent'],
+			];
+			$result = wp_update_post( $args, true );
+			if ( is_wp_error( $result ) ) {
+				return $result;
+			}
+		}
+
+		return parent::update_item( $request );
```

Comment:
Could we simply use `$request['post']` instead? Perhaps like this to work around the parent class limitation:

```php
// WP_REST_Attachments_Controller doesn't allow setting an attachment as the parent post.
// Hence we are working around this here.
$parent_post = ! empty( $request['post'] ) ? (int) $request['post'] : null;
unset( $request['post'] );

$response = parent::update_item( $request );

if ( is_wp_error( $response )  ) {
	return $response;
}

// TODO: Also check for `current_user_can( 'edit_post', (int) $parent_post )` ?
if ( $parent_post && 'attachment' === get_post_type( $parent_post ) ) {
	$args   = [
		'ID'          => $request['id'],
		'post_parent' => $parent_post,
	];
	wp_update_post( $args, true );
}

$attachment = get_post( $request['id'] );

// Ensures that the newly updated post parent is part of the response.
$response = $this->prepare_item_for_response( $attachment, $request );
return rest_ensure_response( $response );
```